### PR TITLE
Added mailer for user registration and letter_opener gem for testing.

### PR DESCRIPTION
**Pull request by maxwellv**
Tests with letter_opener (to open and view the rendered email in the browser) are implemented and working. Right now it is ready for tests via the email_spec gem, though none are written.

### CODE REVIEW

File: spec/mailers/user_mailer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require "spec_helper"
+
+describe UserMailer do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (maxwellv) [inline spec/mailers/user_mailer_spec.rb]: email_spec-based tests should go in this file.

**Comment by maxwellv** [comment]
The email_spec tests are now working. However, the user_signs_up test now fails, though I suspect this is an issue specific to my machine. (It fails because joe@example.com is taken, even though it should work the first time. I think I registered joe@example.com myself and I'm not sure how to delete it from wherever the DB is, or how even to check the DB.)

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class UserMailer < ActionMailer::Base
+  default from: "from@example.com"
+
+  def welcome_email(user)
+    @user = user
+    @url = "http://squmblr.herokuapp.com/"
```
Comment (elizamarcum) [inline app/mailers/user_mailer.rb]: Hard-coding the URL here means that the link will only work for the production deployment.  You should let rails handle this by using root_url in the mail view itself.

File: config/environments/test.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@
   # Tell Action Mailer not to deliver emails to the real world.
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
-  config.action_mailer.delivery_method = :test
+  config.action_mailer.delivery_method = :letter_opener
```
Comment (elizamarcum) [inline config/environments/test.rb]: I think that this should switch back to :test, so that the emails don't always pop open when running the tests.

**Comment by elizamarcum** [comment]
@maxwellv I suspect that pulling in the changes from adding Fabricator will resolve your testing issue.

**Comment by maxwellv** [comment]
I pulled in Fabricator and did further testing on Mitch's machine and found that it worked for him. So yes, I did just somehow pollute my local test DB. It's otherwise working. :shipit:

**Pull request merged**